subscriber/app.py

```diff
@@ -136,6 +136,11 @@ def handle_starttag(self, tag: str, attrs: List[Tuple[str, str | None]]):
             for name, value in attrs:
                 if name == "href" and value is not None:
                     absurl = parse.urljoin(self.base_url, value)
+                    parsed_url = parse.urlparse(absurl)
+
+                    if parsed_url.fragment:
```

Comment:
This is probably good enough for the demo, but it would be even better if we could strip the fragment off the URL in the `get_urls` step.  For example, if `http://example.com/foo` links to `http://example.com/foo/bar#x` and `http://example.com/foo/bar#y`, we should visit `https://example.com/foo/bar` once instead of skipping those URLs altogether.